src/super_gradients/modules/qarepvgg_block.py

```diff
@@ -316,5 +316,16 @@ def fuse_block_residual_branches(self):
     def from_repvgg(self, src: RepVGGBlock):
         raise NotImplementedError
 
-    def prep_model_for_conversion(self, input_size: Optional[Union[tuple, list]] = None, **kwargs):
-        self.partial_fusion()
+    def prep_model_for_conversion(self, input_size: Optional[Union[tuple, list]] = None, full_fusion: bool = True, **kwargs):
```

Comment:
Im wondering whether it would be agood idea to let our users know that once we fuse the model it is not trainable anymore.
On one hand, its not true in the context of predict, on the other hand if someonce calls prep_model_for conversion explicitly- they might try for some reason to train the model again.
Just thinking out loud here.

Reply:
I'd plan for future the following enchancement:
For RepVGGBlock & QARepVGGBlock:
```
class QARepVGGBlock:
   def train():
      if self.fused:
          raise RuntimeError("Fused module cannot be used in training")
```

This would effectively prevent using model with fused blocks in trainer.

Reply:
We can figure it out though

Reply:
Why not:
``` python
class QARepVGGBlock:
   def train():
      if self.full_fusion: # Partially fused model still be trainable for QAT
          raise RuntimeError("Fused module cannot be used in training")
```

Reply:
@BloodAxe Is is part of the scope of this ticket ? https://deci.atlassian.net/browse/SG-839

Reply:
Yup, added it not to forget about it)